shell/app/node_main.cc

```diff
@@ -77,15 +87,27 @@ void ExitIfContainsDisallowedFlags(const std::vector<std::string>& argv) {
   }
 }
 
+#if BUILDFLAG(IS_MAC)
+bool IsInvokedByCurrentApp() {
+  // Get parent process path.
+  // It is possible that proc_pidpath gets denied by macOS when the parent
+  // process is owned by different user, in that case we should just return
+  // false for least privilege.
+  char parent[PROC_PIDPATHINFO_MAXSIZE] = {0};
+  if (proc_pidpath(getppid(), parent, sizeof(parent)) <= 0)
+    return false;
+  // Check if parent process is inside current app's bundle.
+  return base::StartsWith(parent, MainApplicationBundlePath().value() + "/");
```

Comment:
I don't think this check is valid for security purposes, it's not valid as technically speaking anyone can write a file into the resources path and be valid under gatekeeper.  Can we either discuss this here or loop through #wg-security to discuss the intention of this change.  I think the correct thing to do here is to validate the codesignature of the parent process against the designated requirements of this process.

As a security fix it should ideally go through wg-security or the GHSA process so we can do proper analysis.

Reply:
The potential threat this change tries to guard against is, a sandboxed app with limited permissions can hijack an Electron app with more permissions to do more things. For example, malicious `Internet Downloader.app` granted with access only to `~/Downloads` can hijack `VS Code.app` to scan `~/Documents`.

It does not try to guard against attacks from un-sandboxed processes, if a malicious app already has full disk access and can start arbitrary process, then it is not our business.

I don't think this is really a security issue, at least the downstream project did not think it was something that should be fixed in upstream.

@alexdima and @deepak1556 know more about the original issue.

Reply:
Followed-up with details on the wg-security channel.

Reply:
@MarshallOfSound  I have updated the code to do signature validation.